tests/test_appengine_adapter.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""Tests for the AppEngineAdapter."""
+import sys
+
+import mock
+import pytest
+import requests
+
+if requests.__build__ >= 0x021000:
+    from requests.packages.urllib3.contrib import appengine as urllib3_appengine
+    from requests_toolbelt.adapters import appengine
+else:
+    appengine = urllib3_appengine = None
+
+
+@pytest.mark.skipif(sys.version_info >= (3,),
+                    reason="App Engine doesn't support Python 3 (yet)")
```

Comment:
yes, and understood. unfortunately, i think urllib3's existing `AppEngineManager` code may not be python 3 compatible (cc @jonparrott), and i'm not sure i know python 3 well enough myself to fix it, or even to write a test that's portable.

specifically, the `content` attr of `urlfetch.Fetch`'s return value [is a string](https://cloud.google.com/appengine/docs/python/refdocs/google.appengine.api.urlfetch#google.appengine.api.urlfetch.Fetch), which [urllib3 passes to `BytesIO`](https://github.com/snarfed/urllib3/blob/3239c172b19e41f41777db4bd55daae40c99b43d/urllib3/contrib/appengine.py#L178), which wants a `bytes` instead, which doesn't exist in python 2.

full traceback:

```
tests/test_appengine_adapter.py:30:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/py35/lib/python3.5/site-packages/requests/sessions.py:487: in get
    return self.request('GET', url, **kwargs)
.tox/py35/lib/python3.5/site-packages/requests/sessions.py:475: in request
    resp = self.send(prep, **send_kwargs)
.tox/py35/lib/python3.5/site-packages/requests/sessions.py:585: in send
    r = adapter.send(request, **kwargs)
.tox/py35/lib/python3.5/site-packages/requests/adapters.py:403: in send
    timeout=timeout
requests_toolbelt/adapters/appengine.py:117: in urlopen
    **response_kw)
.tox/py35/lib/python3.5/site-packages/requests/packages/urllib3/contrib/appengine.py:136: in urlopen
    response, **response_kw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <requests.packages.urllib3.contrib.appengine.AppEngineManager object at 0x1017ff978>
urlfetch_resp = <Mock id='4320242712'>
response_kw = {'decode_content': False, 'preload_content': False}
transfer_encoding = None

    def _urlfetch_response_to_http_response(self, urlfetch_resp, **response_kw):
    
        if is_prod_appengine():
            # Production GAE handles deflate encoding automatically, but does
            # not remove the encoding header.
            content_encoding = urlfetch_resp.headers.get('content-encoding')
    
            if content_encoding == 'deflate':
                del urlfetch_resp.headers['content-encoding']
    
        transfer_encoding = urlfetch_resp.headers.get('transfer-encoding')
        # We have a full response's content,
        # so let's make sure we don't report ourselves as chunked data.
        if transfer_encoding == 'chunked':
            encodings = transfer_encoding.split(",")
            encodings.remove('chunked')
            urlfetch_resp.headers['transfer-encoding'] = ','.join(encodings)
    
        return HTTPResponse(
            # In order for decoding to work, we must present the content as
            # a file-like object.
>           body=BytesIO(urlfetch_resp.content),
            headers=urlfetch_resp.headers,
            status=urlfetch_resp.status_code,
            **response_kw
        )
       TypeError: a bytes-like object is required, not 'str'

.tox/py35/lib/python3.5/site-packages/requests/packages/urllib3/contrib/appengine.py:178: TypeError
```